### PR TITLE
[10.0] [ADD] sale_order_line_date

### PR DESCRIPTION
**Pull request by nikul-serpentcs**
Sale order line date
===============

This module adds requested date to a sales order lines and propagates it to stock moves and procurements.
When the requested date of the whole sale order is modified the requested date of the lines change to match.

### CODE REVIEW

**Comment by nikul-serpentcs** [comment]
@aheficent Could you please review.

**Comment by LoisRForgeFlow** [comment]
Please, keep commit history always in your migrations. See: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-10.0

**Comment by nikul-serpentcs** [comment]
@lreficent Sure.

**Review by AaronHForgeFlow — APPROVED** [review]
It works good. I'll approve this when the commit history is fixed

**Review by AaronHForgeFlow — CHANGES_REQUESTED** [review]
Until the commit history is fixed

File: sale_order_line_date/models/sale_order.py
Comment (espo-tony) [inline sale_order_line_date/models/sale_order.py]: I think it should also check if 'requested_date' is not already in 'vals'.

**Review by espo-tony — CHANGES_REQUESTED** [review]
It's not clear to me if this module is meant to force lines to have the same requested_date than their order or if it should be possible to set different dates between order and lines.

File: sale_order_line_date/models/sale_order.py
Comment (nikul-serpentcs) [reply sale_order_line_date/models/sale_order.py]: @espo-tony Could you please review.

Comment (espo-tony) [reply sale_order_line_date/models/sale_order.py]: I rather meant is that if requested_date is in vals it shouldn't change its value. In other words, it should only add 'requested_date' in vals if it's not already present.

Is that consistent with the goal of the module?

File: sale_order_line_date/__manifest__.py
Comment (LoisRForgeFlow) [inline sale_order_line_date/__manifest__.py:14]: Should this be changed to `Tecnativa`? @pedrobaeza

**Review by AaronHForgeFlow — COMMENTED** [review]
Hi @nikul-serpentcs can you fix travis? Functional test ok. Thanks.

**Comment by nikul-serpentcs** [comment]
@aheficent Fixed Travis, Could you please review.

**Comment by AaronHForgeFlow** [comment]
Hi @espo-tony, IMHO we should let the user to change the requested date when the SO is in quotation state. Using current approach, if the user makes a mistake there's no way the user can fix it. Furthermore, that is the current behavior in v9.

**Comment by AaronHForgeFlow** [comment]
@espo-tony The requested date in the SO lines is the one that is propagated. The requested date in the SO can be used to set the same requested date to all the SO lines, just that.

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: From my side this is ok.

File: sale_order_line_date/models/sale_order.py
Comment (AaronHForgeFlow) [inline sale_order_line_date/models/sale_order.py]: Hi @nikul-serpentcs I'd put this code in another file called sale_order_line.py to make this easier to read

**Comment by espo-tony** [comment]
Hi @aheficent I totally agree. Indeed it's what I was suggesting: it should only add 'requested_date' in vals if it's not already present - if it's already present in vals, it's because the user manually changed it, and in this case it shouldn't be overwritten.

**Review by espo-tony — APPROVED** [review]
:+1:

**Comment by nikul-serpentcs** [comment]
@aheficent  I did put a code in another file, Could you please review.

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Still working

File: sale_order_line_date/__manifest__.py
Comment (AaronHForgeFlow) [reply sale_order_line_date/__manifest__.py:14]: I think better to change this. @pedrobaeza

**Comment by AaronHForgeFlow** [comment]
@lreficent can you update your review?

**Review by LoisRForgeFlow — APPROVED** [review]
LGTM

**Comment by JordiBForgeFlow** [comment]
@nikul-serpentcs can you squash your commits in order to have a single commit for the migration to v10?

**Comment by nikul-serpentcs** [comment]
@jbeficent squashed commits, Could You Please check it ??

**Review by atchuthan — APPROVED** [review]

**Comment by AaronHForgeFlow** [comment]
IMHO, this can be merged :innocent:

**Pull request merged**
